vesin/src/vesin.cpp

```
@@ -107,7 +98,12 @@ extern "C" void vesin_free(VesinNeighborList* neighbors) {
         // nothing to do
     } else if (neighbors->device == VesinCPU) {
         vesin::cpu::free_neighbors(*neighbors);
+    } else if (neighbors->device == VesinCUDA) {
+        vesin::cuda::free_neighbors(*neighbors);
+    } else {
+        // unknown device
+        std::cerr << "unknown device " << neighbors->device
+                  << " when freeing memory" << std::endl;
```

this should not happen, so at this point we can also throw an error.